merkletree/proof_test.go

```diff
@@ -142,15 +140,25 @@ func TestVerifyProof(t *testing.T) {
 	absentIndex := vrf.Compute([]byte("123"), vrfPrivKey1)
 	proof = m.Get(absentIndex) // shares the same prefix with an empty node
 	verifyProof(t, proof, m.hash, "123")
-	authPathHash(proof)
 	if _, ok := proof.Leaf.(*emptyNode); !ok {
 		t.Error("Invalid proof of absence. Expect an empty node in returned path")
 	}
+}
+
+func TestVerifyProofSamePrefix(t *testing.T) {
+	m, err := NewMerkleTree()
+	if err != nil {
+		t.Fatal(err)
+	}
 
-	/*proof = m.Get([]byte("key4")) // shares the same prefix with leaf node n2
-	verifyProof(t, proof, m.Hash, "key4")
-	authPathHash(proof)
-	if _, ok := proof.Leaf.(*userLeafNode); !ok {
-		t.Error("Invalid proof of absence. Expect a user leaf node in returned path")
-	}*/
+	key1 := "key1"
+	index1 := vrf.Compute([]byte(key1), vrfPrivKey1)
+	val1 := []byte("value1")
+	if err := m.Set(index1, key1, val1); err != nil {
+		t.Fatal(err)
+	}
+	m.recomputeHash()
+	absentIndex := vrf.Compute([]byte("a"), vrfPrivKey1)
+	proof := m.Get(absentIndex) // shares the same prefix with leaf node key1
```

Comment:
Actually `vrfPrivKey1` is deterministic (https://github.com/coniks-sys/coniks-go/blob/master/merkletree/merkletree_test.go#L12-L13). Hope it explains your concern.